### ISSUE TITLE
New saving sessions don't register

### ISSUE THREAD

**Issue by danieltwagner** [BottlecapDave/HomeAssistant-OctopusEnergy#1115]
### Describe the bug

I got an email about an hour ago that new saving sessions are back in order and that one is starting today (2024-12-11) between 6-7pm, along with an opt-in button. I went to check if my automation had fired and noticed that the `binary_sensor.octopus_energy_a_<account>_octoplus_saving_sessions` sensor hadn't changed, and hence the automation didn't run.

I did see that, `octopus_energy_all_octoplus_saving_sessions` shows a bunch of activity, but I'm not sure how to see details for those events.

Could it be that something about the APIs has changed for the 2024 saving sessions? Any other information I can provide?

### Reproduction steps

Be invited to a saving session

### Expected behaviour

Saving session is detected, automation is triggered to join it.

### Tariff Code

20L3168997

### Integration Version

13.2.1

### Home Assistant Version

2024.12.1

### Fresh Install?

Not specified

### Home Assistant Logs

I had not enabled debug logging at the time

### Confirmation

- [X] I confirm that I cannot find my solution within the [documentation](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy)
- [X] I confirm that I cannot find my solution within the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq)

**Comment by JRascagneres** [comment]
Pretty sure this (at the moment) is because the sign up appears to be via email and is not yet on the website / API so there isn't a way for the integration to see the session. Once octopus integrate it into their website this may well work again.

**Comment by plord12** [comment]
Yea ... I created a calendar and updated my automations to trigger from that (as well as the API)

**Comment by BottlecapDave** [comment]
Yeah for some reason they haven't updated the API to include the upcoming event. I hope they're not building yet another set of APIs...

**Comment by JRascagneres** [comment]
@BottlecapDave I think it needs changing as they're now region specific so probably changing things on the backend for that. Its possible that the APIs will change for that.

**Comment by plord12** [comment]
I did wonder if they've gone manual to avoid us automating ....

**Comment by BottlecapDave** [comment]
From what I could gather from their messaging, the regioning seemed like it was superficial leaderboards, but re-reading it, you might be right. Hopefully they can build the free electricity API while they're at it 📟

**Comment by JRascagneres** [comment]
> From what I could gather from their messaging, the regioning seemed like it was superficial leaderboards, but re-reading it, you might be right. Hopefully they can build the free electricity API while they're at it :p

Yeah its not just superficial boards this time. Last year it kinda was for stats.

**Comment by plord12** [comment]
There is also a team name.  I chose "Home Assistant" ... maybe if its near the top Octopus will take more notice :-)

**Comment by johnwb87** [comment]
> > From what I could gather from their messaging, the regioning seemed like it was superficial leaderboards, but re-reading it, you might be right. Hopefully they can build the free electricity API while they're at it :p
> 
> Yeah its not just superficial boards this time. Last year it kinda was for stats. Appears the region is involved in the NESO requirements too.

NESO are now dispatching the DFS as similarly as they can other balancing services. So I’m guessing that the price and utilisation will be better if the aggregator(Octopus for us) can guarantee the region of demand reduction.

**Comment by pjvenda** [comment]
Interestingly, ESO published documentation last year that discussed automation and people 'rigging the system' re the 2022/2023 saving session season. Their comment was that 1. there was no evidence or it wasn't a meaningful proportion and 2. it was not harmful to the purpose of shifting loads. This seemed surprisingly honest and weighted.

I think automation would actually contribute towards the goal. So realistically I wouldn't see why NESO or octopus would actively discourage automating and optimising output of saving sessions. If NESO are paying people it's because they benefit (more) from it. I haven't looked for similar documentation re: last year's activity.

Big thanks to all the devs!

**Comment by scotflieger** [comment]
For previous Saving Sessions I used a Node Red automation to react to the 'Octopus Saving Session Baseline Electricity (acct no)' entity to switch my Tesla Powerwall2s to Off Grid ensuring that I was not consuming any power during the session. Last night I had to do this manually as the entity is no longer present (ie. unknown). As Saving Sessions are now back on it would be useful if the entity could be reinstated if still in the API.

**Comment by danieltwagner** [comment]
Same here, I have an automation (as I imagine many of us do) for load shedding during saving sessions which will turn off non-critical items, such as heating in the workshop. In my opinion this kind of automation is exactly what savings sessions are supposed to encourage and automating it is exactly in line with goals. Let's keep our fingers crossed that lack of API support is just an omission due to time constraints.

**Comment by maegibbons** [comment]
> Yea ... I created a calendar and updated my automations to trigger from that (as well as the API)

Have you got code for this that you can post?

**Comment by gcoan** [comment]
I created some date helper entities (date, start time, end time) and an amount so I could inject those into Predbat as a manual export rate override.

Not fancy but it worked with Predbat deciding whether to discharge my battery or not

**Comment by maegibbons** [comment]
Many thanks!

**Comment by deltamelter** [comment]
I wasn't expecting the autojoin or baselines to work at the moment because OE don't seem to be fully back to using the API.
I had expected to see the latest event in the saving session events entity though.  It only seems to show all the events up until March.   I see this using graphql.  How is the integration fetching SS events? Why can't it see this one?
```
            {
               "id" : 1324,
               "code" : "EVENT_16_111224",
               "startAt" : "2024-12-11T18:00:00+00:00",
               "endAt" : "2024-12-11T19:00:00+00:00",
               "totalParticipants" : 403847,
               "rewardPerKwhInOctoPoints" : 480,
               "devEvent" : false,
               "targetGsp" : [ ],
               "status" : "DONE"
            }
```

**Comment by BottlecapDave** [comment]
I'm seeing the saving session in the attributes of `octoplus_saving_session_events`. The saving sessions are only refetched every 30 minutes and just reports whatever the API reports. I'm hoping the new API isn't _too_ different so that the transition can be quick.

**Comment by gcoan** [comment]
> I'm seeing the saving session in the attributes of `octoplus_saving_session_events`. The saving sessions are only refetched every 30 minutes and just reports whatever the API reports. I'm hoping the new API isn't _too_ different so that the transition can be quick.

I'm seeing the event that was on 11th December now in the joined_events part of the events.  I am pretty sure it wasn't in the list earlier, and it definitely wasn't in the available_events before the session - I could see it wasn't in the event list, and if it was, Predbat would have tried to auto-join it.

Maybe populating the API has only just started?
🤞for the next session

**Comment by deltamelter** [comment]
> I'm seeing the saving session in the attributes of `octoplus_saving_session_events`. The saving sessions are only refetched every 30 minutes and just reports whatever the API reports. I'm hoping the new API isn't _too_ different so that the transition can be quick.

I am still not seeing the event from December in my list.  This is 13.2.1
![image](https://github.com/user-attachments/assets/6be48129-4990-41ea-9fb7-62f960bed80f)

**Comment by deltamelter** [comment]
Oh, this is just showing joined events? I thought it was the complete list of previous events as returned by 
```
query savingsessions($accountNumber: String!){
  savingSessions(accountNumber:  $accountNumber) {
    account(accountNumber: $accountNumber) {
    events 
...
```
![image](https://github.com/user-attachments/assets/89d88471-2f0f-4693-9557-573fcd567f02)
I didn't join the event on 11th.

**Comment by gcoan** [comment]
> Oh, this is just showing joined events? I thought it was the complete list of previous events 

its titled 'joined events' in my event sensor:
![image](https://github.com/user-attachments/assets/19bd78e2-d6dd-4b68-af08-be98acbe2d90)

and the session (which I did join) is showing in the list:
![image](https://github.com/user-attachments/assets/17241b71-5299-4add-808b-b52d6d33f8c5)

Am sure it wasn't earlier.

I exported about 3kWh in the session, so £2 profit or so and the agile rates afterwards were not too bad

**Comment by deltamelter** [comment]
The lists have been reset and only show this season's events now.
so for mein the API, joined_events = [] and events list only shows the 1 event from 11th.
The integration is now just empty so there is no history to show and obviously no future events on the horizon.

**Comment by gcoan** [comment]
Update for second Octopus DFS saving session 8/1/25:

No update !

Today’s DFS saving session hasn’t appeared in either the available or the joined list under the saving sessions event although the prior event from 11/12/24 is still populated as a joined event, with the correct Octopoints that were awarded for my saving included.

The octoplus saving sessions binary sensor is also showing with nothing populated, no current or next joined event details.

So looks like saving sessions are still work in progress Octopus API-land.

**Comment by RobinXe** [comment]
> > > Yea ... I created a calendar and updated my automations to trigger from that (as well as the API)
> > 
> > 
> > Have you got code for this that you can post?
> 
> Its pretty easy. I've got myenergi battery so I want to charge battery before saving session starts, start export when saving session starts and stop when it ends. So 3 automations.
>
>...
> 
> So triggering from both calendar and api.
> 
> Oh, I use a sensor minutes_to_start_of_saving_session so that charge start time depends on the current state of charge (if its near full, start charging late, if its near empty, start charging early).
> 
> This is WIP.

I like your minutes to go idea; I use something similar (though not a sensor, just a templated condition in the automation) to dump battery down to 15% before the start of each Intelligent Octopus overnight period.

If I might make a suggestion; depending on your tariff, it's probably only worth charging the battery up the the state of energy that you are able to export during the period of the saving session. Anything you need to run the house beyond this will either be imported into your battery beforehand, or straight from the grid afterwards, probably at the same price, so sending it through the battery first will cost you a couple of percent more, due to the round-trip losses. This is hardcore min-maxing though, and you may not consider it worth the effort!

**Comment by gcoan** [comment]
yes predbat auto-joined me to the saving session as well and the session is showing in the binary sensor and joined events so looks like Octopus have got the API's working again from last year.

I haven't had the email either. Last year they staggered the notifications to reduce loading on their servers so presumably the same this year

But 24p DFS premium, just not worth it

**Comment by johnwb87** [comment]
Yes, there is a dispatch today. You can find it on the NESO website.

It’s interesting to see that there have actually been a lot of opportunities since the last Saving session and Octopus bid n a favourable price. This is the effect of the changes that we were told about. But also should be keeping the wholesale cost of electricity down compared to the previous premium for DFS

**Comment by BottlecapDave** [comment]
I'm glad they've updated the API without breaking the existing interface. It's hopefully one item off my list. I guess the true test will be when they issue a regional saving session (I'm guessing this is nation wide). I usually auto opt in, but make no real effort to change my habits tbh.

**Comment by johnwb87** [comment]
I can see the event in the event entity, but the binary.sensor entity attributes aren't updating for me.  binary.sensor showing last updated yesterday, other sensors are on the 15 minute cycle as expected.

First opportunity for a predbat auto-join and I'm not there yet.

Event shows in my account as being able to opt-in, don't want to yet so I can diagnose this little issue.

**Comment by funks81** [comment]
So - the saving session today has triggered and opted me in.  Just wondering whether anybody has come up with a condition to stop it opting in based on the number of Octopoints offered?  At 24p - not worth it for me!!

**Comment by noelrobinson** [comment]
Yes my saving session trigger started working today only just noticed, after receiving the email. It means my batteries have now charged ready to dump back to the grid, however at 24p i wish they had not !  Is the reward offered available in the api ahead of time, to make opt in conditional?

**Comment by gcoan** [comment]
> Yes my saving session trigger started working today only just noticed, after receiving the email. It means my batteries have now charged ready to dump back to the grid, however at 24p i wish they had not !  Is the reward offered available in the api ahead of time, to make opt in conditional?

the number of points is in the event https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/entities/octoplus/

**Comment by RobinXe** [comment]
There's no reason to not opt in, as you do not have to participate even once you have. You could certainly use the `octopoints_per_kwh` attribute of the savings session event to determine the behavior of your system during any given event.

**Comment by BottlecapDave** [comment]
As the saving sessions are now being populated again, I'll close this issue. If any further issues arise, please raise a new issue :)
